editor/plugins/tiles/atlas_merging_dialog.cpp

```diff
@@ -104,10 +106,12 @@ void AtlasMergingDialog::_generate_merged(Vector<Ref<TileSetAtlasSource>> p_atla
 				// Create tiles and alternatives, then copy their properties.
 				for (int alternative_index = 0; alternative_index < atlas_source->get_alternative_tiles_count(tile_mapping.key); alternative_index++) {
 					int alternative_id = atlas_source->get_alternative_tile_id(tile_mapping.key, alternative_index);
-					if (alternative_id == 0) {
+					if (alternative_index == 0) {
 						merged->create_tile(tile_mapping.value, atlas_source->get_tile_size_in_atlas(tile_mapping.key));
```

Comment:
@groud Not sure about this change, but the new tile should be created at the first iteration of that loop so when `alternative_index == 0`. Is it guaranteed that `alternative_id` will be `0` at the first iteration or something like that?

Reply:
I did a quick test and alternative tile id at index `0` is always 0. The function works even if there are no alternative tiles, so I assume it's the original, non-alternative tile.

That said, seems like `alternative_id` and `alternative_index` will work the same here.

Reply:
Checked the source of few TileSet methods and it indeed seems it's guaranteed `alternative_id == 0` always exist (for an existing tile) and it's always at index 0 within the alternative ids cache for the tile. So yeah, the previous check should be fine (both should be equivalent), gonna revert this change.